src/adapter/src/coord/dataflows.rs

```diff
@@ -49,14 +51,43 @@ use crate::session::{Session, SERVER_MAJOR_VERSION, SERVER_MINOR_VERSION};
 use crate::util::{viewable_variables, ResultExt};
 use crate::{rbac, AdapterError};
 
+/// A reference-less snapshot of a compute instance.
+#[derive(Debug, Clone)]
+pub struct ComputeInstanceSnapshot {
+    instance_id: ComputeInstanceId,
```

Comment:
naively I'm worried we might run into issues similar to the `ConnectionId` one. i.e. the compute instance doesn't actually exist anymore, but some part of code believes it does since it has a `ComputeInstanceId`. Is this possible? Or do we already assume a compute instance might not exist, even if we have a `ComputeInstanceId`?

Reply:
In the coordinator thread, an instance id won't go away surprisingly since that can also only happen in that thread. Outside of the coord thread, everything has to re-validate when it returns, which happens at the top of each peek stage, which means it'll recheck the instance: https://github.com/MaterializeInc/materialize/blob/cd391bcd33daf9c8651325cc4a28a5cc6ea8f117/src/adapter/src/coord/sequencer/inner.rs#L1992